src/ir/module-utils.cpp

```diff
@@ -590,11 +590,45 @@ namespace {
 
 void classifyTypeVisibility(Module& wasm,
                             InsertOrderedMap<HeapType, HeapTypeInfo>& types) {
+  for (auto type : getPublicHeapTypes(wasm)) {
+    if (auto it = types.find(type); it != types.end()) {
+      it->second.visibility = Visibility::Public;
+    }
+  }
+  for (auto& [type, info] : types) {
+    if (info.visibility != Visibility::Public) {
+      info.visibility = Visibility::Private;
+    }
+  }
+}
+
+void setIndices(IndexedHeapTypes& indexedTypes) {
```

Comment:
Heh, the moved code was so large that it's a smaller diff to show everything else moving around it.